code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,176 @@
+		to_chat(X, span_xenowarning("Our meal is interrupted!"))
+		succeed_activate()
+		add_cooldown()
+		return
+	X.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+	A.emote("scream")
+	A.drop_all_held_items()
+	X.visible_message(A, span_danger("The [X] stabs its tail into [A]!"))
+	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	playsound(A, "alien_claw_flesh", 25, TRUE)
+	X.gain_plasma(X.xeno_caste.drain_plasma_gain)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Rejuvenate/Transfusion
+// ***************************************
+/datum/action/xeno_action/activable/rejuvenate
+	name = "Rejuvenate/Transfusion"
+	action_icon_state = "Rejuvenate"
+	mechanics_text = "When used on self, drains blood and restores health over time. When used on another xenomorph, costs blood and restores some of their health. When used on a dead human, you heal gradually."
+	ability_name = "rejuvenate"
+	cooldown_timer = 2 SECONDS
+	plasma_cost = 0
+	use_state_flags = XACT_TARGET_SELF
+	///Cost of using ability on an ally
+	var/ally_plasma_cost = 20
+	///Cost of using ability on self
+	var/self_plasma_cost = 200
+	///Duration of rejuvenation buff on self
+	var/self_rejuvenation_duration = 5 SECONDS
+
+/datum/action/xeno_action/activable/rejuvenate/can_use_ability(atom/target, silent = FALSE, override_flags) //it is set up to only return true on specific xeno or human targets
+	. = ..()
+	if(!.)
+		return
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
+			to_chat(owner, span_notice("We can only use still, dead hosts to heal."))
+			return FALSE
+		else if(get_dist(owner, H) > 1)
+			to_chat(owner, span_notice("We need to be next to our meal."))
+			return FALSE
+		return TRUE
+	if(isxeno(target))
+		var/mob/living/carbon/xenomorph/gorger/X = owner
+		if(X == target)
+			if(!COOLDOWN_CHECK(X, rejuvenate_self_cooldown))
+				to_chat(X, span_notice("We need another [round(COOLDOWN_TIMELEFT(X, rejuvenate_self_cooldown) / 10)] seconds before we can revitalize ourselves."))
+				return FALSE
+			if(X.plasma_stored < self_plasma_cost)
+				to_chat(X, span_notice("We need [self_plasma_cost - X.plasma_stored]u more blood to revitalize ourselves."))
+				return FALSE
+		else
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				to_chat(X, span_notice("It is beyond our reach, we must be close and our way clear."))
+				return FALSE
+			if(X.plasma_stored < ally_plasma_cost)
+				to_chat(X, span_notice("We need [ally_plasma_cost - X.plasma_stored]u more blood to restore a sister."))
+				return FALSE
+			if(isdead(target))
+				to_chat(X, span_notice("We can only help living sisters."))
+				return FALSE
+			if(!do_mob(X, target, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+				return FALSE
+		return TRUE
```

Comment:
inverte this if/else and you can ealry return

Reply:
changed